grants/search.js

```diff
@@ -239,25 +226,22 @@ async function buildMatchCriteria(queryParams) {
      * Otherwise accept date range in the form yyyy-mm-dd|yyyy-mm-dd
      */
     if (queryParams.awardDate) {
+        let start, end;
         if (/last3months/i.test(queryParams.awardDate)) {
-            const [start, end] = makeDateRange(3);
-
-            match.$and.push({
-                awardDate: { $gte: start, $lt: end }
-            });
+            [start, end] = makeDateRange(3);
         } else if (/last6months/i.test(queryParams.awardDate)) {
-            const [start, end] = makeDateRange(6);
-
-            match.$and.push({
-                awardDate: { $gte: start, $lt: end }
-            });
+            [start, end] = makeDateRange(6);
         } else {
-            const [start, end] = queryParams.awardDate
+            [start, end] = queryParams.awardDate
                 .split('|')
-                .map(str => moment(str).toDate());
+                .map(str => moment(str));
+        }
 
+        if (start && end) {
+            // Ensure the end date is always the last second of the day
+            end = end.set({ hour: 23, minute: 59, second: 59 });
```

Comment:
Added this because the [2004 search](https://www.tnlcommunityfund.org.uk/funding/grants?awardDate=2004-01-01%7C2004-12-31) returns 18,171 results on production, when the actual number is 18,217 (turns out we made a bunch of grants on midnight of the 31st December, which were being marooned in no-man's-land)

Reply:
That earlier data is a bit less reliable date-wise, good catch.